addons/mass_mailing/models/mailing_ab_testing.py

```diff
@@ -0,0 +1,238 @@
+
+    @api.depends('mailing_model_name', 'contact_list_ids')
+    def _compute_mailing_domain(self):
+        for mailing in self:
+            if not mailing.mailing_model_name:
+                mailing.mailing_domain = ''
+            else:
+                mailing.mailing_domain = repr(mailing._get_default_mailing_domain())
+
+    @api.depends('mailing_model_real')
+    def _compute_render_model(self):
+        for mailing in self:
+            mailing.render_model = mailing.mailing_model_real
+
+    @api.depends('mailing_ids')
+    def _compute_mailing_count(self):
+        mailing_data = self.env['mailing.mailing'].read_group(
+            [('testing_mailing_id', 'in', self.ids)],
+            ['testing_mailing_id'], ['testing_mailing_id'])
+        count_data = dict(
+            (item['testing_mailing_id'][0], item['testing_mailing_id_count']) for item in mailing_data
+        )
+        for testing_mailing in self:
+            testing_mailing.mailing_count = count_data.get(testing_mailing.id, 0)
+
+    def _group_expand_states(self, states, domain, order):
+        return [key for key, val in type(self).state.selection]
+
+    def action_add_mailing(self):
+        return {
+            'name': 'Mailing Test',
+            'type': 'ir.actions.act_window',
+            'view_mode': 'form',
+            'res_model': 'mailing.mailing',
+            'context': dict(self._context, **{
+                'default_mailing_model_id': self.mailing_model_id.id,
+                'default_mailing_domain': self.mailing_domain,
+                'default_contact_list_ids': self.contact_list_ids.ids,
+                'default_testing_mailing_id': self.id,
+                'default_mailing_type': self.mailing_type,
+            }),
+        }
+
+    def action_send_mailings(self, mailings=False):
+        self.ensure_one()
+        cron = self.env.ref('mass_mailing.ir_cron_mass_mailing_ab_testing').sudo()
+        if not mailings:
+            mailings = self.mailing_ids.filtered(lambda m: m.state == 'draft')
+        for mailing in mailings:
+            recipients = self._get_recipients(mailing.contact_ab_pc)
+            if recipients:
+                mailing.campaign_id = self.campaign_id
+                mailing.action_send_mail(recipients)
+            else:
+                raise ValidationError(_("There are not enough recipients to send the testing mailings. Increase the sample size or the number of recipients"))
+        if mailing:
+            sent_mailings = self.mailing_ids.filtered(lambda m: m.state == 'done')
+            if sent_mailings == self.mailing_ids and len(self.mailing_ids) >= 2:
+                status = {'state': 'sent'}
+            else:
+                status = {'state': 'in_progress'}
+            self.write(status)
+            at = mailing.sent_date + timedelta(hours=self._get_duration_hours())
+            if not cron.lastcall or at > cron.lastcall:
+                cron._trigger(at=at)
```

Comment:
Not sure this is necessary, lastcall will most likely always be in the past, right?